whois-api/src/main/java/net/ripe/db/whois/api/rdap/WhoisRdapService.java

```diff
@@ -220,17 +276,27 @@ private Query getQueryObject(final Set<ObjectType> objectTypes, final String key
                         key));
     }
 
-    private Response getResponse(HttpServletRequest request, List<RpslObject> result) {
-        if (result.isEmpty()) {
-            throw new NotFoundException("not found");
-        }
+    private Response getOrganisationResponse(HttpServletRequest request, Iterator<RpslObject> result) {
+        return Response.ok(rdapObjectMapper.mapOrganisationEntity(
+                        getRequestUrl(request),
+                        result,
+                        objectDao,
+                        maxEntityResultSize))
+                .header(CONTENT_TYPE, CONTENT_TYPE_RDAP_JSON)
+                .build();
+    }
+    private Response getResponse(HttpServletRequest request, Iterable<RpslObject> result) {
+        Iterator<RpslObject> rpslIterator = result.iterator();
 
-        if (result.size() > 1) {
-            throw new IllegalStateException("Unexpected result size: " + result.size());
+        if (!rpslIterator.hasNext()) {
+            throw new NotFoundException("not found");
         }
 
+        final RpslObject resultObject = rpslIterator.next();
 
-        final RpslObject resultObject = result.get(0);
+        if (rpslIterator.hasNext()) {
```

Comment:
so there's only ever a single RpslObject returned? if so then better name this method `getSingleObjectReponse`